ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -330,6 +330,13 @@ public void analyze(ASTNode ast, Context ctx) throws SemanticException {
     initCtx(ctx);
     init(true);
     analyzeInternal(ast);
+    setQueryPropertiesAfterAnalyze();
+  }
+
+  private void setQueryPropertiesAfterAnalyze() {
+    if (queryProperties != null && getQB() != null){
+      queryProperties.extractInfoFromQueryBlock(getQB());
```

Comment:
why not just add this in `analyzeInternal`, naming is wierd containing the sequence order
also should it be `queryProperties.addInfoFromQueryBlock`?

Reply:
ready to change naming
the issue with ordering is that we have tons of semantic analyzer subclasses, and they tend to inherit analyzerInternal
as I want to be independent of them, I must run this afterwards...setQueryPropertiesAfterAnalyze is indeed weird, what about changing it to:
```
private void postAnalyze()
```
adding javadoc also